### ISSUE TITLE
Unable to launch windows processes from WSL when using long environment variables

### ISSUE THREAD

**Issue by jamesthurley** [microsoft/WSL#7947]
### Version

Windows version: 10.0.22000.434 / WSL version: 0.50.2.0 / WSLg version: 1.0.29

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.74.3

### Distro Version

Ubuntu 20.04

### Other Software

git version 2.34.1.windows.1
VS Code version 1.63.2

### Repro Steps

Between each of these scenarios I do the following:
 - restart Windows Terminal
 - run `wsl --shutdown` 
 - open a linux terminal
 - run a command such as `code .` or, in a repo which requires the credential manager to be used, `git pull`.

Note some of these environment variables are too long to use `setx`. Use the windows UI or `[Environment]::SetEnvironmentVariable`.

## Scenario 1: _WSLENV long, TEST_VARIABLE long._

Set `WSLENV` environment variable (long):
```
TEST_VARIABLE:VARIABLE_1:VARIABLE_2:VARIABLE_3:VARIABLE_4:VARIABLE_5:VARIABLE_6:VARIABLE_7:VARIABLE_8:VARIABLE_9:VARIABLE_10:VARIABLE_11:VARIABLE_12:VARIABLE_13:VARIABLE_14:VARIABLE_15:VARIABLE_16:VARIABLE_17:VARIABLE_18:VARIABLE_19:VARIABLE_20:VARIABLE_21:VARIABLE_22:VARIABLE_23:VARIABLE_24:VARIABLE_25:VARIABLE_26:VARIABLE_27:VARIABLE_28:VARIABLE_29:VARIABLE_30:VARIABLE_31:VARIABLE_32:VARIABLE_33:VARIABLE_34:VARIABLE_35:VARIABLE_36:VARIABLE_37:VARIABLE_38:VARIABLE_39:VARIABLE_40:VARIABLE_41:VARIABLE_42:VARIABLE_43:VARIABLE_44:VARIABLE_45:VARIABLE_46:VARIABLE_47:VARIABLE_48:VARIABLE_49:VARIABLE_50:VARIABLE_51:VARIABLE_52:VARIABLE_53:VARIABLE_54:VARIABLE_55:VARIABLE_56:VARIABLE_57:VARIABLE_58:VARIABLE_59:VARIABLE_60:VARIABLE_61:VARIABLE_62:VARIABLE_63:VARIABLE_64:VARIABLE_65:VARIABLE_66:VARIABLE_64:VARIABLE_68:VARIABLE_69:VARIABLE_70:VARIABLE_71:VARIABLE_72:VARIABLE_73:VARIABLE_74:VARIABLE_75:VARIABLE_76:VARIABLE_77:X_VALID_REGIONS:WT_PROFILE_ID:WT_SESSION
```

Set `TEST_VARIABLE` environment variable to (long, 2048 chars):
```
There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with an
```

## Scenario 2: _WSLENV long, TEST_VARIABLE short._

Set `WSLENV` environment variable (long):
```
TEST_VARIABLE:VARIABLE_1:VARIABLE_2:VARIABLE_3:VARIABLE_4:VARIABLE_5:VARIABLE_6:VARIABLE_7:VARIABLE_8:VARIABLE_9:VARIABLE_10:VARIABLE_11:VARIABLE_12:VARIABLE_13:VARIABLE_14:VARIABLE_15:VARIABLE_16:VARIABLE_17:VARIABLE_18:VARIABLE_19:VARIABLE_20:VARIABLE_21:VARIABLE_22:VARIABLE_23:VARIABLE_24:VARIABLE_25:VARIABLE_26:VARIABLE_27:VARIABLE_28:VARIABLE_29:VARIABLE_30:VARIABLE_31:VARIABLE_32:VARIABLE_33:VARIABLE_34:VARIABLE_35:VARIABLE_36:VARIABLE_37:VARIABLE_38:VARIABLE_39:VARIABLE_40:VARIABLE_41:VARIABLE_42:VARIABLE_43:VARIABLE_44:VARIABLE_45:VARIABLE_46:VARIABLE_47:VARIABLE_48:VARIABLE_49:VARIABLE_50:VARIABLE_51:VARIABLE_52:VARIABLE_53:VARIABLE_54:VARIABLE_55:VARIABLE_56:VARIABLE_57:VARIABLE_58:VARIABLE_59:VARIABLE_60:VARIABLE_61:VARIABLE_62:VARIABLE_63:VARIABLE_64:VARIABLE_65:VARIABLE_66:VARIABLE_64:VARIABLE_68:VARIABLE_69:VARIABLE_70:VARIABLE_71:VARIABLE_72:VARIABLE_73:VARIABLE_74:VARIABLE_75:VARIABLE_76:VARIABLE_77:X_VALID_REGIONS:WT_PROFILE_ID:WT_SESSION
```

Set `TEST_VARIABLE` environment variable to (short):
```
There are many variations
```

## Scenario 3: _WSLENV short, TEST_VARIABLE long._

Set `WSLENV` environment variable (short):
```
TEST_VARIABLE:VARIABLE_1:VARIABLE_2:VARIABLE_3:X_VALID_REGIONS:WT_PROFILE_ID:WT_SESSION
```

Set `TEST_VARIABLE` environment variable to (long, 2048 chars):
```
There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with an
```

### Expected Behavior

All three scenarios should allow you to launch VS Code, or interact with git using the windows credential manager.

### Actual Behavior

Scenario 1 fails:

```
$ code .
<3>init: (30) ERROR: UtilAcceptVsock:238: accept4 failed 110
```

```
$ GIT_TRACE=1 git pull
17:38:15.747118 git.c:439               trace: built-in: git pull
17:38:15.747329 run-command.c:663       trace: run_command: git fetch --update-head-ok
17:38:15.748323 git.c:439               trace: built-in: git fetch --update-head-ok
17:38:15.748776 run-command.c:663       trace: run_command: GIT_DIR=.git git-remote-https origin https://github.com/CanopySimulations/canopy-admin-tools.git
17:38:16.140650 run-command.c:663       trace: run_command: '/mnt/c/Program\ Files/Git/mingw64/libexec/git-core/git-credential-manager-core.exe get'
<3>init: (44) ERROR: UtilAcceptVsock:238: accept4 failed 110
Username for 'https://github.com':
```

In scenarios 2 or 3, where one of the two variables is significantly shorter, everything works as expected. VS Code will launch, and git will be able to authenticate.

This is as far as I've narrowed it down so far, but if I can be of any more help debugging please let me know.  I think this should be reproducible as I reformatted my (Windows 10) PC due what I now think was this same problem a few days ago, and then encountered it again on the fresh install of Windows 11. Thankfully starting with a fresh windows installation allowed me to identify the environment variables as the issue.

**Comment by jamesthurley** [comment]
Also, on my old Windows 10 installation the behaviour was that trying to launch VS Code would hang forever, and the same with the git credential manager. 

On Windows 11 above the behaviour changed slightly to displaying the error message `<3>init: (44) ERROR: UtilAcceptVsock:238: accept4 failed 110`.

However I think they were the same issue as I started seeing the problem in Windows 10 around the time I started having a lot of items in my `WSLENV` variable.
